common/logging/src/tracing_logging_layer.rs

```diff
@@ -0,0 +1,137 @@
+use std::collections::HashMap;
+use std::fs::{self, create_dir_all, File, OpenOptions};
+use std::io::Write;
+use std::path::PathBuf;
+use std::sync::mpsc::{self, Sender};
+use std::thread;
+
+lazy_static! {
+    pub static ref TRACING_LOGGING_DEPENDENCIES: Vec<String> =
+        vec!["libp2p".to_string(), "discv5".to_string()];
```

Comment:
note that none of the `libp2p` dependencies has a target/module path named libp2p. I hacked my changes to get metrics on debug to showcase some of the names we will actually get
```
dep_debug_total{target="libp2p_core"} 11
dep_debug_total{target="libp2p_dns"} 8
dep_debug_total{target="libp2p_gossipsub"} 273
dep_debug_total{target="libp2p_mplex"} 31
dep_debug_total{target="libp2p_quic"} 2
dep_deubg_total{target="libp2p_swarm"} 15
dep_debug_total{target="libp2p_tcp"} 13
```
we probably need to decide which of these we want and what to do with enabled logs of deps not on the list @AgeManning

Reply:
I'm now capturing libp2p_gossipsub, but can make changes to include some, or all, of the libp2p dependencies based on what we decide

Reply:
Yeah I think just gossipsub is fine initially. We can maybe add some others in later if things get really spicy.